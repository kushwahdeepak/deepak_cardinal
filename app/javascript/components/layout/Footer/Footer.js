import React from 'react'
import Navbar from 'react-bootstrap/Navbar'
import Nav from 'react-bootstrap/Nav'
import Row from 'react-bootstrap/Row'
import Image from 'react-bootstrap/Image'
import NavLink from 'react-bootstrap/NavLink'

import styles from './styles/Footer.module.scss'
import LogoWithText from '../../../../assets/images/logos/navbar-logo.svg'

const Footer = () => {
    return (
        <Row
            className={`justify-content-between align-items-center py-3 px-3 mx-0 ${styles.footerContainer}`}
        >
            <Navbar  className="d-flex flex-column py-0 w-100">
                <Navbar.Toggle />
                <Navbar.Collapse
                    id="basic-navbar-nav"
                    className={`w-100 flex-column`}
                >
                    <div className={styles.footerLinks} >
                        <Navbar.Brand href="/" className="d-none d-lg-block">
                            <Image
                                src={LogoWithText}
                                rounded
                                className={styles.footerLogo}
                            />
                        </Navbar.Brand>
                        <div className="d-flex flex-column">
                            <p className={'mb-0 ' + styles.footerLink}>
                                COMPANY
                            </p>
                            <a
                                href="/welcome/about_us"
                                className={styles.footerLink}
                            >
                                About us
                            </a>
                            <a
                                href="/welcome/careers"
                                className={styles.footerLink}
                            >
                                Cardinal Talent Careers
                            </a>
                        </div>
                    </div>
                    <div className={`d-flex ${styles.privacyP}`}>
                        <NavLink
                            href="/privacy_policy"
                            className={styles.footerLink}
                        >
                            Privacy policy
                        </NavLink>
                        <span className={styles.footerpip}>|</span>
                        <NavLink
                            href="/terms_of_service"
                            className={styles.footerLink}
                        >
                            Terms of service
                        </NavLink>
                    </div>
                </Navbar.Collapse>
            </Navbar>
        </Row>
    )
}

export default Footer
