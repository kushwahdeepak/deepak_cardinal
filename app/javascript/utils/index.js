import { act } from 'react-dom/test-utils'

export const findByTestAttr = (wrapper, attr) => {
    return wrapper.find(`[data-test='${attr}']`)
}

export const timeoutResolve = (time = 0, response = {}) =>
    new Promise((resolve) => {
        setTimeout(() => resolve(response), time)
    })

export const wait = (time = 0) => {
    const timer = timeoutResolve(time)

    return act(async () => {
        await timer
    })
}
export function emailIsValid(email) {
    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)
}

export function generateRandom(min, max) {
    return Math.floor(min + Math.random() * (max + 1 - min))
}

export function capitalizeFirstLetter(string) {
    return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase()
}

export function createMarkedList(array) {
    if (!Array.isArray(array)) {
        return []
    }
    return array.map((candidate) => ({
        id: candidate.id,
        name: candidate.id,
        fullName: `${candidate.first_name} ${candidate.last_name}`,
        emailAddress: candidate.email_address,
        isChecked: false,
    }))
}

export function stringReplace(string,toSymbol,fromSymbol){
    return string.replace(toSymbol, fromSymbol)
}

export function capitalize(string) {
    return string.replace(/\b(\w)/g, s => s.toUpperCase());
}

export function skillIsNotEmpty(value){
    return value.trim().length > 0;
}

export function firstCharacterCapital(value){
    return value?.charAt(0)?.toUpperCase() + value?.slice(1)
}
