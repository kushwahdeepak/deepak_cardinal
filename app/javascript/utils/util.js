import isNil from 'lodash.isnil'
import isEmpty from 'lodash.isempty'

class Util {
    /**
     * Define stage values
     */
    static STAGES = ['first_interview', 'second_interview', 'offer', 'reject']

    /**
     * Tests whether a single object attribute includes the at
     * least one string from the given set.
     */
    static attributePassesStringSetFilter(object, attr, stringSet) {
        if (isNil(object[attr])) return false

        // We "OR" the tag tests: the attribute needs to contain only one tag
        return stringSet
            .map((string) => string.trim().toLowerCase())
            .reduce(
                (passes, string) =>
                    passes ||
                    Util.attributeToString(object[attr])
                        .toLowerCase()
                        .includes(string),
                false
            )
    }

    /**
     * Tests whether at least one of a set of object attributes
     * includes every word from a given piece of text.
     */
    static objectPassesTextFilter(object, attrs, text) {
        text = text.trim().toLowerCase()

        if (text === '') return true

        const words = text.split(' ')

        // Join all the given attribute names to lowercase, stringified attribute values
        const allAttrsString = attrs
            .filter((attr) => !isNil(object[attr]))
            .map((attr) => Util.attributeToString(object[attr]).toLowerCase())
            .join('')

        // Is every word included in the allAttrsString?
        return words.reduce(
            (matchesAll, word) => matchesAll && allAttrsString.includes(word),
            true
        )
    }

    static attributeToString(attr) {
        if (typeof attr === 'string') {
            return attr
        } else if (Array.isArray(attr)) {
            return attr.join(',')
        } else if (typeof attr === 'object') {
            return JSON.stringify(attr)
        } else {
            return '' + attr
        }
    }

    static numberWithCommas(x) {
        return x?.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
    }

    /**
     * Returns number of results in format [x] - [y] of [z]
     * @param {number} totalJobsCount Total number of jobs
     * @param {number} pageCount Total number of pages
     * @param {number} activePage Currently active page
     * @param {number} pageSize How many results to display per page
     */
    static displayNumberOfResults(
        totalJobsCount,
        pageCount,
        activePage,
        pageSize,
        totalPersonCount=null,
        candidatePage=false,
        showdisplaycount=true
    ) {
        if (totalJobsCount == 0) return '0 - 0 of 0'
        const offset = pageSize * pageCount - totalJobsCount
        const lastJobIndex =
            (activePage + 1) * pageSize > totalJobsCount
                ? totalJobsCount
                : (activePage + 1) * pageSize
        const firstJobIndex =
            activePage == 0
                ? 1
                : activePage == pageCount - 1
                ? lastJobIndex - pageSize + offset + 1
                : lastJobIndex - pageSize + 1
        return `${firstJobIndex} - ${lastJobIndex} of ${Util.numberWithCommas(
            (candidatePage && showdisplaycount) ? totalPersonCount : totalJobsCount

        )}`
    }

    /**
     * Returns a string indicating the candidate's title and current
     * company, based on available information on the Person object
     * @param {object} candidate Person object from our db
     */
    static candidateCompanyString(candidate) {
        const position = candidate.company_position
        const currentCompany = candidate.current_company

        if (isEmpty(position) && isEmpty(currentCompany)) return ''

        if (!isEmpty(position) && !isEmpty(currentCompany))
            return `${position} at ${currentCompany}`

        if (!isEmpty(position)) return position

        if (!isEmpty(currentCompany)) return `at ${currentCompany}`
    }

    /**
     * Converts the given array of objects with ids to an object whose
     * keys are the ids of the array objects.
     *
     * Every object in the given array must have a unique 'id' property defined.
     *
     * @param {Array} array
     * @returns {Object}
     */
    static arrayToObject(array) {
        const obj = {}
        array.forEach((element) => (obj[element.id] = element))
        return obj
    }

    /**
     * Makes a copy of the given array and replace the value at 'index' with 'value'
     *
     * @param {Array} array
     * @param {Number} index
     * @param {*} value
     * @returns {Array}
     */
    static replaceInArrayCopy_i(array, index, value) {
        const clone = array.slice()
        clone[index] = value
        return clone
    }

    /**
     * Makes a copy of the given array and replaces the element with the same
     * .id property as obj.id with obj
     *
     * @param {Array} array
     * @param {Object} obj
     * @returns {Array}
     */
    static replaceInArrayCopy(array, obj) {
        const index = array.findIndex((element) => element.id === obj.id)
        return Util.replaceInArrayCopy_i(array, index, obj)
    }

    static deleteInArrayCopy(array, index) {
        const copy = array.slice()
        copy.splice(index, 1)

        return copy
    }

    static handleUndefined(value){
        if (value === undefined || value === 'undefined')
        return '';
        else
        return value
    }

    static handleUndefinedFullName(firstName, lastName){
      if(firstName === "undefined" || firstName === undefined)
      return lastName
      else if(lastName === "undefined" || lastName === undefined)
      return firstName
      else
      return firstName + ' ' + lastName     }

    static capitalize(value){
         return value.charAt(0).toUpperCase() + value.slice(1).toLowerCase();
    }
}

export default Util
