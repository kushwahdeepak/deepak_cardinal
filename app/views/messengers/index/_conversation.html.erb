<!-- erbstart app/views/messengers/index/_conversation.html.erb -->
<div class="col-sm-10">
  <div class="conversation">
    <div class="start-conversation">
      <div>
        <i class="fa fa-inbox" aria-hidden="true"></i><br>
        Open a conversation
      </div>
    </div>
  </div>
</div>
<!-- erbend app/views/messengers/index/_conversation.html.erb -->
