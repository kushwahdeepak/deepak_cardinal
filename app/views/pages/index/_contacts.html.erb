<!-- erbstart app/views/pages/index/_contacts.html.erb -->
<div id="contacts">
  <h4>Contacts</h4>
  <ul>
    <% @contacts.each do |contact| %>
      <%= render 'pages/index/contacts/contact', contact: contact %>
    <% end %>
  </ul>
  <%= render no_contacts_partial_path %>
</div>
<!-- erbend app/views/pages/index/_contacts.html.erb -->
