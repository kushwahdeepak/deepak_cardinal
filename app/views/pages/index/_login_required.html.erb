<!-- erbstart app/views/pages/index/_login_required.html.erb -->
<div class="non-signed-in-message">
  <%= link_to 'Login', 
              login_path, 
              class: 'login-button-index' %> to access all features
</div>
<!-- erbend app/views/pages/index/_login_required.html.erb -->
