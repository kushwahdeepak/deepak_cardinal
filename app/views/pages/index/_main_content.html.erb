<!-- erbstart app/views/pages/index/_main_content.html.erb -->
<div id="main-content" class="col-sm-9">
  <h3 class="page-name"><%= link_to 'job_type', job_type_posts_path %></h3>
  <div class="row">
    <%= render @job_type_posts %>
    <%= render no_posts_partial_path(@job_type_posts) %>
  </div><!-- row -->

  <h3 class="page-name"><%= link_to 'Study', study_posts_path %></h3>
  <div class="row">
    <%= render @study_posts %>
    <%= render no_posts_partial_path(@study_posts) %>
  </div><!-- row -->

  <h3 class="page-name"><%= link_to 'Team member', team_posts_path %></h3>
  <div class="row">
    <%= render @team_posts %>
    <%= render no_posts_partial_path(@team_posts) %>
  </div><!-- row -->
</div><!-- main_content -->
<!-- erbend app/views/pages/index/_main_content.html.erb -->
