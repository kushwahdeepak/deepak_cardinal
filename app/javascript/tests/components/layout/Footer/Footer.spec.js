import React from 'react'
import Footer from '../../../../components/layout/Footer/Footer'
import renderer from 'react-test-renderer'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { findByTestAttr } from '../../../../utils/index'

configure({ adapter: new Adapter() })

describe('Footer component', function () {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<Footer />)
    })

    it('should render Footer component', () => {
        expect(wrapper.exists()).toBe(true)
    })
    it('should match with a snapshot', () => {
        const component = renderer.create(<Footer />)
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('should contain the root footer tag', () => {
        expect(wrapper.find('footer').length).toBe(1)
    })
    it('should contain the correct number of children in the root tag', () => {
        expect(wrapper.find('footer').children().length).toBe(1)
    })
    it('should contain correct href prop for links', () => {
        expect(
            findByTestAttr(wrapper, 'report-bug-link').at(0).props().href
        ).toBe('mailto:tech-support@cardinalhire.com')
    })
})
