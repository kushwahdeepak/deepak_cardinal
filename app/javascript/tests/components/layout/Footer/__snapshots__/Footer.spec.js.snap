// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should match with a snapshot 1`] = `
<footer>
  <div
    className="justify-content-between py-3 px-5 row"
  >
    <img
      className="rounded"
      src="test-file-stub"
    />
    <ul>
      <li>
        Contact Us
      </li>
      <li>
        CT@CardinalTalent.ai
      </li>
      <li>
        (xxx)xxx-xxx
      </li>
      <li>
        <a
          data-test="report-bug-link"
          href="mailto:tech-support@cardinalhire.com"
          target="_blank"
        >
          Report a bug
        </a>
      </li>
    </ul>
  </div>
</footer>
`;
